src/main/java/org/octri/hpoonfhir/controller/MainController.java

```diff
@@ -1,27 +1,78 @@
 package org.octri.hpoonfhir.controller;
 
+import java.util.List;
 import java.util.Map;
+import java.util.stream.Collectors;
 
 import javax.servlet.http.HttpServletRequest;
 
+import org.hl7.fhir.dstu3.model.Patient;
+import org.hl7.fhir.exceptions.FHIRException;
+import org.monarchinitiative.fhir2hpo.loinc.DefaultLoinc2HpoAnnotation;
 import org.monarchinitiative.fhir2hpo.loinc.Loinc2HpoAnnotation;
 import org.monarchinitiative.fhir2hpo.loinc.LoincId;
 import org.monarchinitiative.fhir2hpo.service.AnnotationService;
+import org.octri.hpoonfhir.service.FhirService;
+import org.octri.hpoonfhir.view.PatientModel;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.bind.annotation.RequestMapping;
 
 @Controller
 public class MainController {
-	
+
 	@Autowired
 	AnnotationService annotationService;
+	
+	@Autowired
+	@Qualifier("r3FhirService")
+	FhirService fhirService;
 
-	@RequestMapping("/")
-	public String search(Map<String, Object> model, HttpServletRequest request) {
+	/**
+	 * Return to a clean search form with no results.
+	 * @param model
+	 * @param request
+	 * @return
+	 */
+	@GetMapping("/")
+	public String home(Map<String, Object> model, HttpServletRequest request) {
+		model.put("patientSearchForm", new PatientModel());
+		model.put("results", false);
+		return "search";
+	}
+	
+	/**
+	 * Search using the form parameters, and return results.
+	 * @param model
+	 * @param form
+	 * @return
+	 */
+	@PostMapping("/search")
+	public String search(Map<String, Object> model, @ModelAttribute PatientModel form) {
+		model.put("patientSearchForm", form);
+		try {
+			List<Patient> patients = fhirService.findPatientsByFullName(form.getFirstName(), form.getLastName());
+			List<PatientModel> patientModels = patients.stream().map(fhirPatient -> {
+				PatientModel patientModel = new PatientModel();
+				patientModel.setFirstName(fhirPatient.getNameFirstRep().getGivenAsSingleString());
+				patientModel.setLastName(fhirPatient.getNameFirstRep().getFamily());
+				patientModel.setId(fhirPatient.getIdElement().getIdPart());
+				return patientModel;
```

Comment:
Alternatively, you could create a constructor in the PatientModel class that initializes using a Patient parameter.